viking/src/tools/check.rs

```diff
@@ -488,15 +497,67 @@ fn parse_args() -> Result<Args, lexopt::Error> {
         version,
         always_diff,
         update_matching,
+        print_help,
         other_args,
     })
 }
 
+fn print_help() -> Result<()> {
+    println!(
+"Usage: check [function name] [--version VERSION] [--update-matching] [--always-diff] [asm-differ arguments]
+    
+Checks if the compiled bytecode of a function matches the assembly found within the game elf. If not, show the differences between them.
+If no function name is provided, all functions within the respository function list will be checked.
+
+optional arguments:
+
+ -h, --help             Show this help message and exit
+ --version VERSION      Check the function against version VERSION of the game elf
+ --update-matching      If a function does match, but is marked as mismatching, mark it as matching in the function list
+ --always-diff          Show an assembly diff, even if the function matches
+All further arguments are forwarded onto asm-differ. 
+
+asm-differ arguments:"
+);
+
+    let differ_path = repo::get_tools_path()?.join("asm-differ").join("diff.py");
+
+    // By default, invoking asm-differ using std::process:Process doesn't seem to allow argparse
+    // (the python module asm-differ uses to print its help text) to correctly determine the number of columns in the host terminal.
+    // To work around this, we'll detect that for it, and set it manually via the COLUMNS environment variable
+    let num_columns = match termsize::get() {
+        Some(size) => size.cols,
```

Comment:
[crossterm](https://docs.rs/crossterm/0.13.1/crossterm/index.html)::[terminal](https://docs.rs/crossterm/0.13.1/crossterm/terminal/index.html)::[size](https://docs.rs/crossterm/0.13.1/crossterm/terminal/fn.size.html) can be used to get the terminal size -- no need for a separate terminal library.

Reply:
Oh, fantastic - I'll switch to that and remove the additional dependency.